### PR TITLE
OPRUN-3063: Generate OLM microshift manifests

### PR DESCRIPTION
**Pull request by jmprusi**
This PR adds the necessary changes to 'scripts/generate_crds_manifests.sh' to create a new set of manifests specifically targeted for microshift.

### CODE REVIEW

**Comment by ncdc** [comment]
@jmprusi please split this into 2 commits - 1 for the script edits, 1 for the generated files. Thanks!

File: scripts/generate_crds_manifests.sh [hunk 1]
```diff
@@ -497,3 +494,52 @@ find "${ROOT_DIR}/manifests" -type f -exec $SED -i "1{/---/d}" {} \;
...
+# Create the kustomization file
+cat << EOF > "${ROOT_DIR}/microshift-manifests/kustomization.yaml"
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+resources:
+EOF
+
+# Now let's generate the kustomization.yaml file for microshift to pick up the manifests.
+microshift_manifests_files=$(find "${ROOT_DIR}/microshift-manifests" -type f -name "*.yaml")
+# filest to ignore, substring match.
```
Comment (ncdc) [inline scripts/generate_crds_manifests.sh:524]:
```suggestion
# files to ignore, substring match.
```

**Comment by jmprusi** [comment]
@ncdc done, PTAL.

**Comment by ncdc** [comment]
Need to fix verify failure

**Comment by jmprusi** [comment]
> Need to fix verify failure

`make verify` returns 0 locally. looks like we are having some CI issues...

**Comment by ncdc** [comment]
It's complaining about kustomization.yaml

**Comment by jmprusi** [comment]
> It's complaining about kustomization.yaml

Fixed... It looks like `find` ordering is different in Macos vs Linux.

File: scripts/generate_crds_manifests.sh [hunk 2]
```diff
@@ -495,3 +492,54 @@ find "${ROOT_DIR}/manifests" -type f -exec $SED -i "1{/---/d}" {} \;
...
 ${YQ} delete --inplace -d'*' manifests/0000_50_olm_00-namespace.yaml 'metadata.labels."pod-security.kubernetes.io/audit*"'
+
+
+# Let's copy all the manifests to a separate directory for microshift
+mkdir -p "${ROOT_DIR}/microshift-manifests/"
+rm -rf "${ROOT_DIR}/microshift-manifests/"*
+cp "${ROOT_DIR}"/manifests/* "${ROOT_DIR}/microshift-manifests/"
+
+# Let's add the annotations for ibm-cloud-managed on the standard manifests
+# !! Order matters, as we don't need the IBM managed annotations on the microshift manifests.
```
Comment (ncdc) [inline scripts/generate_crds_manifests.sh:503]: What would happen if we did have these annotations on the manifests used by MicroShift? I'm thinking the less we differ, the better?

File: microshift-manifests/0000_50_olm_00-pprof-config.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  annotations:
+    include.release.openshift.io/ibm-cloud-managed: "true"
+    include.release.openshift.io/self-managed-high-availability: "true"
+    release.openshift.io/create-only: "true"
```
Comment (ncdc) [inline microshift-manifests/0000_50_olm_00-pprof-config.yaml:7]: This is an example of 1 file where the regular manifest has `capability.openshift.io/name: "OperatorLifecycleManager"` but the MicroShift variant does not. I think this capability annotation should be on the MicroShift manifests too?

File: scripts/generate_crds_manifests.sh (hunk 2 above)
Comment (jmprusi) [reply scripts/generate_crds_manifests.sh]: done, PTAL

File: microshift-manifests/0000_50_olm_00-pprof-config.yaml (hunk 3 above)
Comment (jmprusi) [reply microshift-manifests/0000_50_olm_00-pprof-config.yaml:7]: done, PTAL

**Pull request merged**